### PR TITLE
Add SegFormer ONNX support

### PR DESCRIPTION
**Pull request by NielsRogge**
# What does this PR do?

This PR adds support for the first "semantic-segmentation" feature, SegFormer.

Related to #16308

### CODE REVIEW

File: src/transformers/models/segformer/configuration_segformer.py
Comment (regisss) [inline src/transformers/models/segformer/configuration_segformer.py:166]: The output for `semantic-segmentation` defined in `src/transformers/onnx/config.py` has dynamic axes for height and width, but here the input does not. Have you tested this works with inputs of different sizes?

**Review by lewtun — APPROVED** [review]
Thanks for adding this ONNX export @NielsRogge ! 

Apart from the comment by @regisss (which I agree should be checked), this PR LGTM. Have you also tested that the ONNX slow tests pass for this model:\

```
RUN_SLOW=1 pytest tests/onnx/test_onnx_v2.py -k "segformer"
```

File: src/transformers/models/segformer/configuration_segformer.py
Comment (lewtun) [reply src/transformers/models/segformer/configuration_segformer.py:166]: I agree - we should probably have something like the following:
```suggestion
                ("pixel_values", {0: "batch", 1: "num_labels", 2: "height", 3: "width"}),
```

See the "improved" `semantic-segmentation` feature from https://github.com/huggingface/transformers/pull/18587 - this will be merged today

File: src/transformers/onnx/features.py
Comment (lewtun) [inline src/transformers/onnx/features.py:421]: FYI @ydshieh discovered that the `adaptive_avg_pool2d` is not currently supported in ONNX. Not sure if that affects SegFormer or not, but if yes, you'll want to mimic what we did for Data2Vec: https://github.com/huggingface/transformers/pull/18587/files#diff-f4bad33d844c5d91b09dd0af27395ade640e907920b357daf125dd19458bdd60R243

Comment (ydshieh) [reply src/transformers/onnx/features.py:421]: PS: Not found any string `adaptive` in `modeling_segformer.py` file.

File: src/transformers/models/segformer/configuration_segformer.py
Comment (NielsRogge) [reply src/transformers/models/segformer/configuration_segformer.py:166]: Do you mean `num_channels` instead of `num_labels`?

Cause `num_labels` is relevant for the output logits, but not for the inputs.

**Comment by NielsRogge** [comment]
@lewtun I have confirmed that all tests pass!

**Comment by NielsRogge** [comment]
@lewtun ok for me to merge?

File: src/transformers/models/segformer/configuration_segformer.py
Comment (lewtun) [reply src/transformers/models/segformer/configuration_segformer.py]: Oh yes, I meant `num_channels` :)

Your implementation LGTM! 

Let's wait for final approval from @sgugger before merging

**Review by sgugger — APPROVED** [review]
Thanks for adding this!

**Pull request merged**
